panic: runtime error: invalid memory address or nil pointer dereference [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x26db554]

[https://jenkins.vega.rocks/job/common/job/system-tests-nightly/46/testReport/junit/tes[…]ll_tests___full_a_f___test_datanode_restore_from_dehistory/](https://jenkins.vega.rocks/job/common/job/system-tests-nightly/46/testReport/junit/tests.datanode/datanode_test/Call_tests___full_a_f___test_datanode_restore_from_dehistory/)

```
Trace from file: /jenkins/workspace/common/system-tests-wrapper/networkdata/testnet/logs/testnet-nodeset-full-7-full/vega-full-7.stderr-2023-09-30T02:38:28Z.log

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x26db554]

goroutine 144 [running]:
github.com/tendermint/tendermint/light/provider/http.(*http).LightBlock(0xc0000d3b40, {0x5b7a798, 0xc000e11db0}, 0x0)
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/light/provider/http/http.go:86 +0x134
github.com/tendermint/tendermint/light.(*Client).getTargetBlockOrLatest(0xc0000d3b40?, {0x5b7a798, 0xc000e11db0}, 0x249c, {0x5b72c60, 0xc0000d3b40})
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/light/detector.go:384 +0x3c
github.com/tendermint/tendermint/light.(*Client).compareNewHeaderWithWitness(0xc000e7a900, {0x5b7a798, 0xc000e11db0}, 0x0?, 0xc0012cf398, {0x5b72c60, 0xc0000d3b40}, 0x5)
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/light/detector.go:138 +0x27a
created by github.com/tendermint/tendermint/light.(*Client).compareFirstHeaderWithWitnesses
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/cometbft@v0.34.28-0.20230322133204-3d8588de736e/light/client.go:1144 +0x1bd

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -23,12 +23,12 @@
 - [8741](https://github.com/vegaprotocol/vega/issues/8741) - Add a network parameter for disabling `Ethereum` oracles.
 - [8600](https://github.com/vegaprotocol/vega/issues/8600) - Clean and refactor data source packages.
 - [8845](https://github.com/vegaprotocol/vega/issues/8845) - Add support for network treasury and global insurance accounts.
-- [9545](https://github.com/vegaprotocol/vega/issues/9545) - Auto load new segments after load finishes 
+- [9545](https://github.com/vegaprotocol/vega/issues/9545) - Auto load new segments after load finishes
 - [8661](https://github.com/vegaprotocol/vega/issues/8661) - Refactor the snapshot engine to make it testable.
 - [8680](https://github.com/vegaprotocol/vega/issues/8680) - Move loading the local snapshot in the initialization steps.
 - [8682](https://github.com/vegaprotocol/vega/issues/8682) - Share snapshot by search the metadata database instead of loading the tree.
 - [8846](https://github.com/vegaprotocol/vega/issues/8846) - Add support to transfer recurring transfers to metric based reward
-- [9549](https://github.com/vegaprotocol/vega/issues/9549) - Update config defaults to better support archive nodes 
+- [9549](https://github.com/vegaprotocol/vega/issues/9549) - Update config defaults to better support archive nodes
 - [8857](https://github.com/vegaprotocol/vega/issues/8857) - Add a step for getting the balance of the liquidity provider liquidity fee account
 - [8847](https://github.com/vegaprotocol/vega/issues/8847) - Implement internal time trigger data source.
 - [8895](https://github.com/vegaprotocol/vega/issues/8895) - Allow to set runtime parameters in the SQL Store connection structure
@@ -122,6 +122,7 @@
 - [9590](https://github.com/vegaprotocol/vega/issues/9590) - Restore positions for market activity tracker on migration from old version.
 - [9589](https://github.com/vegaprotocol/vega/issues/9589) - Add event for funding payments.
 - [9460](https://github.com/vegaprotocol/vega/issues/9460) - Add APIs for volume discount program.
+- [9628](https://github.com/vegaprotocol/vega/issues/9628) - Upgrade `CometBFT`.
 
 ### 🐛 Fixes
 
@@ -139,7 +140,7 @@
 - [9343](https://github.com/vegaprotocol/vega/issues/9343) - Prevent malicious validator submitting Ethereum oracle chain event prior to initial start time
 - [8792](https://github.com/vegaprotocol/vega/issues/8792) - Fix panic when starting null block chain node.
 - [8739](https://github.com/vegaprotocol/vega/issues/8739) - Cancel orders for rejected markets.
-- [9594](https://github.com/vegaprotocol/vega/issues/9594) - non deterministic order events on market termination 
+- [9594](https://github.com/vegaprotocol/vega/issues/9594) - non deterministic order events on market termination
 - [9350](https://github.com/vegaprotocol/vega/issues/9350) - Clear account ledger events causing segment divergence
 - [9118](https://github.com/vegaprotocol/vega/issues/9118) - Improve list stop orders error message
 - [9406](https://github.com/vegaprotocol/vega/issues/9105) - Fix Ethereum oracle validation failing unexpectedly when using go 1.19

--- a/go.mod
+++ b/go.mod
@@ -100,7 +100,7 @@ require (
 	bazil.org/fuse v0.0.0-20200117225306-7b5117fecadc // indirect
 	contrib.go.opencensus.io/exporter/prometheus v0.4.2 // indirect
 	github.com/AndreasBriese/bbloom v0.0.0-20190825152654-46b345b51c96 // indirect
-	github.com/ChainSafe/go-schnorrkel v0.0.0-20200405005733-88cbf1b4c40d // indirect
+	github.com/ChainSafe/go-schnorrkel v1.0.0 // indirect
 	github.com/DataDog/zstd v1.5.2 // indirect
 	github.com/VictoriaMetrics/fastcache v1.6.0 // indirect
 	github.com/Workiva/go-datastructures v1.0.53 // indirect
@@ -381,6 +381,6 @@ replace (
 	github.com/fergusstrange/embedded-postgres => github.com/vegaprotocol/embedded-postgres v1.13.1-0.20221123183204-2e7a2feee5bb
 	github.com/muesli/cancelreader => github.com/vegaprotocol/cancelreader v0.0.0-20230724130739-6f2217a69449
 	github.com/shopspring/decimal => github.com/vegaprotocol/decimal v1.3.1-uint256
-	github.com/tendermint/tendermint => github.com/vegaprotocol/cometbft v0.34.28-0.20230322133204-3d8588de736e
+	github.com/tendermint/tendermint => github.com/vegaprotocol/cometbft v0.34.29-patch.1
 	github.com/tendermint/tm-db => github.com/cometbft/cometbft-db v0.6.7
 )

